dbux-projects/src/dataLib/PathwaysDataProvider.js

```diff
@@ -68,30 +70,36 @@ class ApplicationCollection extends PathwaysCollection {
    * @param {Array<Application>} applications 
    */
   postAddRaw(applications) {
-    for (const app of applications) {
-      const filePath = this.dp.session.getApplicationFilePath(app.uuid);
-      const { version, collections } = app.dataProvider.serializeJson();
-      const header = JSON.stringify({ headerTag: true, version });
-      if (!fs.existsSync(filePath)) {
-        try {
-          const s = JSON.stringify(collections);
-          // const s = Object.entries(collections || EmptyObject).map(([key, value]) => ).join(',');
-          fs.appendFileSync(filePath, `${header}\n${s}\n`, { flag: 'ax' });
-        }
-        catch (err) {
-          logError(`Cannot write header of application log file at ${filePath}. Error:`, err);
-        }
-        app.dataProvider.onAnyData(data => {
-          const { collections: serializedNewData } = app.dataProvider.serializeJson(Object.entries(data));
-          fs.appendFileSync(filePath, JSON.stringify(serializedNewData) + '\n');
-        });
-      }
-      else {
-        warn(`Found existing log file at "${filePath}", added old application without \`isNew=false\``);
-      }
+    this.handleApplicationAdded(applications);
+    this.dp.updateIndexFile();
+  }
+
+  handleApplicationAdded(apps) {
+    for (const app of apps) {
+      app.dataProvider.onAnyData(data => {
+        this.handleDataAdded(app, data);
+      });
+
+      this.handleDataAdded(app);
     }
+  }
 
-    this.dp.updateIndexFile();
+  handleDataAdded(app, allData = null) {
```

Comment:
Can we add a debounce to this (or else it might run a whole bunch of times in quick succession when new data comes in, once per collection, and maybe even more often)?

Reply:
NOTE: The data we are writting depends on the paramater. If we want to apply debounce on this, we will need to "merge" the data manually. Skipped for now.